rero/rero-ils
# templates: implement record templates

A new resource `template` is added for the records of
types document, patron, item and holdings.

A non-controlled field called `data` in the template
json schema is available for librarians to save
pre-validated data.

A template can have two visibilities either:
1. public, available for use for all users of the
organisation.
2. private, available only for use for its creator.

Users with role librarian can create private templates,
 edit their own private records, and access public and
their own private records of their organisation.

Users with role system_librarian can create/edit/access/delete
any type of templates of its organisation.

* Adds pids to the users.json fixtures data file.
* Adds units testing and fixtures for the new resource.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/us/1642?milestone=271177

## How to test?

`bootstrap + setup`
login as `asrtid`, you can see some templates here ~/api/templates/ 

## Code review check list
